### ISSUE TITLE
Compilation error: redeclaration of friend may not have default template args

### ISSUE THREAD

**Issue by omaralvarez** [mariusbancila/croncpp#1]
I am having the following compilation error:

```
In file included from /mnt/data/compile/croncpp_orig/benchmark/main.cpp:7:0:
/mnt/data/compile/croncpp_orig/include/croncpp.h:713:20: error: redeclaration of friend ‘template<class Traits> cron::cronexpr cron::make_cron(std::string_view)’ may not have default template arguments
    static cronexpr make_cron(std::string_view expr)
                    ^~~~~~~~~
In file included from /mnt/data/compile/croncpp_orig/benchmark/main.cpp:7:0:
/mnt/data/compile/croncpp_orig/include/croncpp.h:24:20: note: ‘template<class Traits> cron::cronexpr cron::make_cron(std::string_view)’ previously declared here
    static cronexpr make_cron(std::string_view expr);
                    ^~~~~~~~~
/mnt/data/compile/croncpp_orig/benchmark/main.cpp: In function ‘void test_cron(const std::vector<std::__cxx11::basic_string<char> >&)’:
/mnt/data/compile/croncpp_orig/benchmark/main.cpp:32:39: error: no matching function for call to ‘make_cron(const std::__cxx11::basic_string<char>&)’
          auto cron = cron::make_cron(e);
                                       ^
In file included from /mnt/data/compile/croncpp_orig/benchmark/main.cpp:7:0:
/mnt/data/compile/croncpp_orig/include/croncpp.h:24:20: note: candidate: template<class Traits> cron::cronexpr cron::make_cron(std::string_view)
    static cronexpr make_cron(std::string_view expr);
                    ^~~~~~~~~
/mnt/data/compile/croncpp_orig/include/croncpp.h:24:20: note:   template argument deduction/substitution failed:
/mnt/data/compile/croncpp_orig/benchmark/main.cpp:32:39: note:   couldn't deduce template parameter ‘Traits’
          auto cron = cron::make_cron(e);
```
I'm using Ubuntu 18.04 and gcc 7.3.0, is this normal? Issue with my gcc?

**Comment by omaralvarez** [comment]
Thanks for the quick answer. I will test it right away. I just removed the default template type, and it compiled fine.

**Comment by omaralvarez** [comment]
I have just tested it and it works perfectly. I have forked your lib to add CMake install targets in order to be able to `make install` the library. It's a WIP.

https://github.com/omaralvarez/croncpp

I also created:

https://github.com/omaralvarez/conan-croncpp

And also:

https://bintray.com/omaralvarez/public-conan/croncpp%3Aomaralvarez

So that the library can easily be installed with Conan.

Let me know if you want a pull request.
